vllm/entrypoints/openai/serving_engine.py

```diff
@@ -187,6 +192,8 @@ def _validate_prompt_and_tokenize(
         else:
             input_ids = prompt_ids
 
+        input_text = prompt if prompt is not None else self.tokenizer.decode(
+            prompt_ids)
```

Comment:
I don't think we should be unconditionally detokenizing the prompt ids in this case. It would be better to do this if/where it's actually needed. At minimum only if echo is specified.